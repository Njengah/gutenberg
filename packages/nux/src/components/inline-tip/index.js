/**
 * WordPress dependencies
 */
import { Notice } from '@wordpress/components';
import { compose } from '@wordpress/compose';
import { withSelect, withDispatch } from '@wordpress/data';
import { __ } from '@wordpress/i18n';

export function InlineTip( { isVisible, className, onRemove, children } ) {
	if ( ! isVisible ) {
		return null;
	}

	return (
		<Notice className={ className } onRemove={ onRemove }>
			{ children }
		</Notice>
	);
}

export default compose(
	withSelect( ( select, { tipId } ) => ( {
		isVisible: select( 'core/nux' ).isTipVisible( tipId ),
	} ) ),
	withDispatch( ( dispatch, { tipId } ) => {
		const { disableTips, dismissTip } = dispatch( 'core/nux' );

		return {
			onRemove() {
				// Disable reason: We don't yet have a <Confirm> component. One day!
				// eslint-disable-next-line no-alert
				if ( window.confirm( __( 'Would you like to disable tips like these in the future? ' ) ) ) {
					disableTips();
				} else {
					dismissTip( tipId );
				}
			},
		};
	} )
)( InlineTip );
