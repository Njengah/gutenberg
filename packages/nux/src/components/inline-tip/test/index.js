/**
 * External dependencies
 */
import TestRenderer from 'react-test-renderer';

/**
 * WordPress dependencies
 */
import { Notice } from '@wordpress/components';

/**
 * Internal dependencies
 */
import { InlineTip } from '../';

describe( 'InlineTip', () => {
	it( 'should not render anything if invisible', () => {
		const renderer = TestRenderer.create(
			<InlineTip isVisible={ false }>
				It looks like you’re writing a letter. Would you like help?
			</InlineTip>
		);
		expect( renderer.root.children ).toHaveLength( 0 );
	} );

	it( 'should render correctly', () => {
		const renderer = TestRenderer.create(
			<InlineTip isVisible>
				It looks like you’re writing a letter. Would you like help?
			</InlineTip>
		);
		expect( renderer ).toMatchSnapshot();
	} );

	it( 'calls `onRemove` when the rendered notice is dismissed', () => {
		const onRemove = jest.fn();
		const renderer = TestRenderer.create(
			<InlineTip isVisible onRemove={ onRemove }>
				It looks like you’re writing a letter. Would you like help?
			</InlineTip>
		);
		renderer.root.findByType( Notice ).props.onRemove();
		expect( onRemove ).toHaveBeenCalled();
	} );
} );
