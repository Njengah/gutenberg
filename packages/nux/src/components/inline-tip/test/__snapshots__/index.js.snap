// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineTip should render correctly 1`] = `
<div
  className="components-notice is-undefined is-dismissible"
>
  <div
    className="components-notice__content"
  >
    It looks like youâ€™re writing a letter. Would you like help?
  </div>
  <button
    aria-label="Dismiss this notice"
    className="components-button components-icon-button components-notice__dismiss"
    onClick={[Function]}
    type="button"
  >
    <svg
      aria-hidden="true"
      className="dashicon dashicons-no"
      focusable="false"
      height={20}
      role="img"
      viewBox="0 0 20 20"
      width={20}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12.12 10l3.53 3.53-2.12 2.12L10 12.12l-3.54 3.54-2.12-2.12L7.88 10 4.34 6.46l2.12-2.12L10 7.88l3.54-3.53 2.12 2.12z"
      />
    </svg>
  </button>
</div>
`;
